CMakeLists.txt

```diff
@@ -74,14 +76,41 @@ if (NOT WIN32) # If on windows, do not build binaries that do not support window
   target_link_libraries(edlib-aligner edlib)
 endif()
 
-
-# Create target 'install' for installing libraries.
-install(TARGETS edlib DESTINATION ${CMAKE_INSTALL_LIBDIR})
-install(FILES edlib/include/edlib.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
-
 # configure and install pkg-config file
 configure_file(
     ${CMAKE_CURRENT_SOURCE_DIR}/edlib.pc.in
     ${CMAKE_CURRENT_BINARY_DIR}/edlib-1.pc
     @ONLY)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/edlib-1.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+
+if(EDLIB_ENABLE_INSTALL)
+	install(FILES ${CMAKE_CURRENT_BINARY_DIR}/edlib-1.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
+	include(CMakePackageConfigHelpers)
+	set(EDLIB_CMAKE_DIR "${CMAKE_INSTALL_LIBDIR}/cmake/edlib" CACHE STRING
+	  "Installation directory for cmake files, relative to ${CMAKE_INSTALL_PREFIX}.")
+	set(version_config "${PROJECT_BINARY_DIR}/edlib-config-version.cmake")
+	set(project_config "${PROJECT_BINARY_DIR}/edlib-config.cmake")
+	set(targets_export_name edlib-targets)
+
+	# Generate the version, config and target files into the build directory.
+	write_basic_package_version_file(
+		${version_config}
+		VERSION ${VERSION}
+		COMPATIBILITY AnyNewerVersion)
+	configure_package_config_file(
+		${PROJECT_SOURCE_DIR}/edlib-config.cmake.in
+		${project_config}
+		INSTALL_DESTINATION ${EDLIB_CMAKE_DIR})
+	export(TARGETS edlib NAMESPACE edlib::
+		FILE ${PROJECT_BINARY_DIR}/${targets_export_name}.cmake)
+
+	# Install version, config and target files.
```

Comment:
Could you just mention that installing/producing these metadata files enables tools like vcpkg to manipulate/use Edlib?